### ISSUE TITLE
Distinguish between connection problem and missing Python support

### ISSUE THREAD

**Issue by duck-rh** [ansible/ansible#28870]
##### ISSUE TYPE
 - Bug Report

##### COMPONENT NAME
connection

##### ANSIBLE VERSION
```
ansible 2.3.1.0
  config file = /home/duck/myproject/ansible.cfg
  configured module search path = Default w/o overrides
  python version = 2.7.13+ (default, Jul 19 2017, 18:15:03) [GCC 6.4.0 20170704]
```

##### CONFIGURATION
The managed host is a Vagrant host, accessible via SSH on a specific port with sudo. But this is not important, see below.

##### OS / ENVIRONMENT
I'm managing a Fedora 26 host using a Debian unstable controller.

##### SUMMARY
When the connection fails because the Python binary cannot be found, Ansible should say so instead of blaming the connection.

##### STEPS TO REPRODUCE
Any playbook would do. The only important thing is to have an OS which has no Python binary installed or has switched to Python 3 (like Fedora 26 in this case).

##### EXPECTED RESULTS
Ansible should tell the connection is ok but it could not find the Python binary.

##### ACTUAL RESULTS
```
TASK [Gathering Facts] ****************************************************************************************************************************
Using module file /home/duck/.local/lib/python2.7/site-packages/ansible/modules/system/setup.py
<127.0.0.1> ESTABLISH SSH CONNECTION FOR USER: vagrant
<127.0.0.1> SSH: EXEC ssh -vvv -o UserKnownHostsFile=/dev/null -o IdentitiesOnly=yes -i /home/duck/.vagrant.d/insecure_private_key -o ControlMaster=auto -o ControlPersist=60s -o StrictHostKeyChecking=no -o Port=10022 -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=vagrant -o ConnectTimeout=30 -o ControlPath=/home/duck/.ansible/cp/1c53231ee0 127.0.0.1 '/bin/sh -c '"'"'sudo -H -S  -p "[sudo via ansible, key=gsagkhjfbovoospitvliggzzeujpkyjt] password: " -u root /bin/sh -c '"'"'"'"'"'"'"'"'echo BECOME-SUCCESS-gsagkhjfbovoospitvliggzzeujpkyjt; /usr/bin/python'"'"'"'"'"'"'"'"' && sleep 0'"'"''
fatal: [ovirt-web-builder.osci.io]: UNREACHABLE! => {
    "changed": false, 
    "msg": "SSH Error: data could not be sent to remote host \"127.0.0.1\". Make sure this host can be reached over ssh", 
    "unreachable": true
}
```

Setting _ansible_python_interpreter_ solved my problem but after I needlessly investigated a connection problem. In the verbose output we can see Ansible is running the echo and call to Python in the same SSH call, but this needs to be decoupled.

**Comment by samdoran** [comment]
Thank you for reporting this issue. As you pointed out, we would need to decouple  the `ssh` call from the Python call in order to do accurate error reporting, which is a non-trivial undertaking. We don't have plans to so at this time.

**Comment by duck-rh** [comment]
@samdoran  I can perfectly understand you don't have time for this now, but I can't understand why you closed the bug. The problem is still there and should be tracked.

**Comment by samdoran** [comment]
Point taken. I well reopen but not sure when we will get to this. Thank you so much for bringing this up and digging enough to find the root cause.

**Comment by abadger** [comment]
!python3

**Comment by bcoca** [comment]
fixed via #43576

**Comment by sowbug** [comment]
This happened to me, too, after installing Raspbian Lite on a new machine. Wasted a lot of time debugging the ssh configuration.
